projects/ama-sdk/src/lib/helpers/services/error-handler.service.ts

```diff
@@ -0,0 +1,44 @@
+ /*!
+ * @license
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ErrorHandler, Injectable, Injector } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { ErrorService } from './error.service';
+import { NotificationService } from './notification.service';
+
+@Injectable()
+export class GlobalErrorHandler implements ErrorHandler {
+
+    constructor(private injector: Injector) { }
+
+    handleError(error: Error | HttpErrorResponse) {
+
+        const errorService = this.injector.get(ErrorService);
```

Comment:
Why injecting it using the injector? Can't we use the normal DI injection by angular (in the constructor)?